packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -92,6 +93,184 @@ export interface StageOptions extends StageProps {
   readonly placement?: StagePlacement;
 }
 
+/**
+ * Pipeline types.
+ */
+export enum PipelineType {
+  /**
+   * V1 type
+   */
+  V1 = 'V1',
+  /**
+   * V2 type
+   */
+  V2 = 'V2',
+}
+
+/**
+ * Properties of pipeline-level variable.
+ */
+export interface VariableProps {
```

Comment:
Lets put variable code into `variable.ts` and trigger code into `trigger.ts`

Reply:
I separated the files and test files.

https://github.com/aws/aws-cdk/pull/28538/commits/1b4bc2108900f464811cce7cad75acdb357fb5e6

Reply:
The trigger is an interface, not a class. Is it okay if I have them in separate files?